components/DailyCrusher/DailyCrusher.js

```diff
@@ -0,0 +1,171 @@
+      6: workouts[5],
+      0: workouts[0],
+    };
+
+    return workoutSchedule[today] || null;
+  }
+
+  const currentWorkout = getCurrentDayWorkout();
+
+  if (!currentWorkout) {
+    return <p>No workout scheduled for today. Rest day!</p>;
+  }
+
+  const muscleGroups = [];
+
+  const enrichedExercises = currentWorkout.exercises.map((exercise) => {
+    const foundExercise = findExerciseById(exercises, exercise.exerciseId);
+    if (foundExercise) {
+      muscleGroups.push(...foundExercise.muscleGroups);
+      return { ...exercise, name: foundExercise.name };
+    }
+    return exercise;
+  });
+
+  const uniqueMuscleGroups = [...new Set(muscleGroups)];
+
+  const dailyCrusher = {
+    ...currentWorkout,
+    exercises: enrichedExercises,
+    muscleGroups: uniqueMuscleGroups,
+  };
+
+  return (
+    <StyledCrusherWrapper>
+      <StyledHeadline>Crusher of the Day 💪</StyledHeadline>
+      <StyledArticleWrap>
+        <StyledDescriptionText>
+          Get in shape with our crusher workouts! Every day we suggest a
+          different workout for you to try out.
+        </StyledDescriptionText>
+
+        <StyledLink href="/workoutlist">
+          <Workout workout={dailyCrusher} spotlightMode={true} />
+        </StyledLink>
+        <StyledPostionRelative>
+          {isCrusherDetailsVisible ? null : (
+            <StyledAbsoluteToggleButton onClick={toggleVisibility}>
+              Show More
+            </StyledAbsoluteToggleButton>
+          )}
+          <StyledPreviewFilter
+            $showMore={
+              isCrusherDetailsVisible ? "filter: none; color: black;" : null
+            }
+          >
+            <h3>{dailyCrusher.title}</h3>
+            <StyledDescriptionText>
+              {dailyCrusher.description1}
+            </StyledDescriptionText>
+          </StyledPreviewFilter>
+        </StyledPostionRelative>
+        {isCrusherDetailsVisible ? (
+          <>
+            {dailyCrusher.exercises.map((exercise) => {
+              return (
+                <StyledExerciseSetsReps key={uid()}>
+                  <StyledBoldText>{exercise.name}: </StyledBoldText> Sets:{" "}
+                  {exercise.sets} / Reps: {exercise.reps}
+                </StyledExerciseSetsReps>
+              );
+            })}
+            <StyledDescriptionText>
+              {dailyCrusher.description2}
+            </StyledDescriptionText>
+            <StyledDescriptionText>
+              {dailyCrusher.description3}
+            </StyledDescriptionText>
+          </>
+        ) : null}
+      </StyledArticleWrap>
+      {!isCrusherDetailsVisible ? null : (
+        <StyledToggleButton onClick={toggleVisibility}>
+          Show Less
+        </StyledToggleButton>
+      )}
+    </StyledCrusherWrapper>
+  );
+}
+
+const StyledCrusherWrapper = styled.article`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  border-radius: 12px;
+  max-width: 600px;
+  margin: 20px auto;
```

Comment:
We were told in the course that we should not use any margin at all if possible. So we try to put containers around it and give them padding when necessary.

Reply:
Thank you for this advice :)